src/bindings/kotlin/templates/ObjectTemplate.kt

```diff
@@ -1,31 +1,25 @@
+
 class {{ obj.name()|class_name_kt }}(handle: Long) {
     private var handle: AtomicLong = AtomicLong(handle)
     {%- for cons in obj.constructors() %}
-    constructor({% for arg in cons.arguments() %}{{ arg.name() }}: {{ arg.type_()|type_kt }}{% if loop.last %}{% else %}, {% endif %}{% endfor %}) :
-        this(
-            _UniFFILib.INSTANCE.{{ cons.ffi_func().name() }}(
-                {%- for arg in cons.arguments() %}
-                {{ arg.name()|lower_kt(arg.type_()) }}{% if loop.last %}{% else %},{% endif %}
-                {%- endfor %}
-            )
-        )
+    constructor({% call kt::arg_list_decl(cons.arguments()) -%}) :
+        this({% call kt::to_rs_call(cons) %})
     {%- endfor %}
 
     // XXX TODO: destructors or equivalent.
 
-    {%- for meth in obj.methods() %}
-    fun {{ meth.name()|fn_name_kt }}(
-        {% for arg in meth.arguments() %}
-        {{ arg.name() }}: {{ arg.type_()|type_kt }}{% if loop.last %}{% else %}, {% endif %}
-        {% endfor %}
-    ): {% match meth.return_type() %}{% when Some with (type_) %}{{ type_|type_kt }}{% when None %}Unit{% endmatch %} {
-        val _retval = _UniFFILib.INSTANCE.{{ meth.ffi_func().name() }}(
-            this.handle.get(){% if meth.arguments().len() > 0 %},{% endif %}
-            {%- for arg in meth.arguments() %}
-            {{ arg.name()|lower_kt(arg.type_()) }}{% if loop.last %}{% else %},{% endif %}
-            {%- endfor %}
-        )
-        {% match meth.return_type() %}{% when Some with (return_type) %}return {{ "_retval"|lift_kt(return_type) }}{% else %}return _retval{% endmatch %}
+    {% for meth in obj.methods() -%}
+    {%- match meth.return_type() -%}
+
+    {%- when Some with (return_type) -%}
+    fun {{ meth.name()|fn_name_kt }}({% call kt::arg_list_decl(meth.arguments()) %}): {{ return_type|type_kt }} {
+        val _retval = {% call kt::to_rs_call_with_prefix("this.handle.get()", meth) %}
+        return {{ "_retval"|lift_kt(return_type) }}
     }
-    {%- endfor %}
+    
+    {%- when None -%}
+    fun {{ meth.name()|fn_name_kt }}({% call kt::arg_list_decl(meth.arguments()) %}) =
+        {% call kt::to_rs_call_with_prefix("this.handle.get()", meth) %}
+    {% endmatch %}
```

Comment:
This ended up a lot more readable switching to the doing the match around the whole method (like we do in `TopLevelFunctionTemplate.kt`) instead of doing it twice and/or a clunky `return _retVal` when `_retVal` was `Unit`.